### PR TITLE
Add StatsUpsellModal

### PR DESCRIPTION
**Pull request by lsl**
Fixes https://github.com/Automattic/dotcom-forge/issues/4911

cc @crisbusquets 

## Proposed Changes

* Adds the StatsUpsellModal

## Testing Instructions

* Visit the calypso live link, append the query param to the url enable the paywall gating for the session:  `?flags=stats/paid-wpcom-v2`
* e.g. https://container-nifty-newton.calypso.live/?flags=stats/paid-wpcom-v2
* Click on a stats specific upgrade button under /stats

Todo:
- [x] Confirm we're still going with a personal upsell
- [x] Fix plan pricing and localization
- [x] Localized currency font size needs reducing to match figma
- [x] Check tracks events are firing right
- [x] Check checkout link redirect to works and ideally update it to go back to the current path instead of just /stats/day/:site

Doing in follow up prs:
- [ ] Use it for the download csv button
- [ ] Use it for the banner button
- [ ] If using premium, account for upgrades from personal in the pricing being displayed - moved: https://github.com/Automattic/dotcom-forge/issues/4997

Screenshot 2023-12-19 at 17-05-08 Jetpack Stats ‹ Site Title — WordPress com

### CODE REVIEW

File: client/my-sites/stats/stats-upsell-modal/index.tsx
Comment (lsl) [inline client/my-sites/stats/stats-upsell-modal/index.tsx:47]: Needed to use this in order to match the design / avoid this header section:

Screenshot 2023-12-19 at 14-39-06 Jetpack Stats ‹ Site Title — WordPress com

**Comment by crisbusquets** [comment]
Hi @lsl,
How should I test this? I couldn't trigger the modal using the Calypso.live links. Or should I wait? :)

**Comment by lsl** [comment]
@crisbusquets Sorry, meant to mention the config flag. Testing instructions updated.

**Comment by crisbusquets** [comment]
> Calypso Live [(direct link)](https://calypso.live?image=registry.a8c.com/calypso/app:build-91909)
> Jetpack Cloud live [(direct link)](https://calypso.live?image=registry.a8c.com/calypso/app:build-91909&env=jetpack)

Thanks @lsl !

There are a few things that need adjustments compared to Figma:
* Modal width: 700px instead of 860px
* Line-height for the heading should be 40px (instead of 48px)
* Checkmark icon, 18x18 instead of 24x24

**Comment by allilevine** [comment]
@lsl Adding Cris' comment from Slack:

> what about adding this last item to the list (All Explorer plan features)?

p1703089221518989/1703086267.795209-slack-C068X5XJT3J

**Comment by lsl** [comment]
> There are a few things that need adjustments compared to Figma:

Done

> what about adding this last item to the list (All Explorer plan features)?

Done

@crisbusquets any idea why the currency font seems bigger here than on figma? I think I got the sizing right but it looks noticeably larger. I also had to center the line vertically as aligning to the top made it misaligned.

**Review by candy02058912 — COMMENTED** [review]
This is what the modal looks like on smaller screen sizes
medium-sized (is that how I call this?)
mobile-modal

small-sized
mobile

I think we need a design review from @crisbusquets, but it won't be blocking this PR since it's all behind a feature flag.

File: client/my-sites/stats/stats-upsell-modal/index.tsx
Comment (candy02058912) [inline client/my-sites/stats/stats-upsell-modal/index.tsx:89]: Haven't played with code too much yet, am curious why we're using `dangerouslySetInnerHTML` here?

File: client/my-sites/stats/stats-upsell-modal/index.tsx
Comment (candy02058912) [inline client/my-sites/stats/stats-upsell-modal/index.tsx:100]: Seeing a couple of these `stats-upsell-modal__feature` being repeated, what do you think about mapping through them, for example:
```
const features = [
  { key: 'all_stats', text: 'All stats available: traffic trends, sources, optimal time to post…' },
  // ...
];
```
And then:
```
{features.map((feature) => (
  <div key={feature.key} className="stats-upsell-modal__feature">
    <Gridicon icon="checkmark" size={18} />
    <div className="stats-upsell-modal__feature-text">{translate(feature.text)}</div>
  </div>
))}
```

**Comment by crisbusquets** [comment]
> This is what the modal looks like on smaller screen sizes medium-sized (is that how I call this?) mobile-modal
> 
> small-sized mobile
> 
> I think we need a design review from @crisbusquets, but it won't be blocking this PR since it's all behind a feature flag.

I'll work on this, @candy02058912. And share a proposal asap.

File: client/my-sites/stats/stats-upsell-modal/index.tsx
Comment (candy02058912) [inline client/my-sites/stats/stats-upsell-modal/index.tsx]: Getting this Tracks error due to the naming:
```
Tracks: Event `stats_upsell_modal_submit` will be ignored because it does not match /^calypso(?:_[a-z0-9]+){2,}$/ nor /^jetpack(?:_[a-z0-9]+){2,}$/ and is not a listed exception. Please use a compliant event name.
```

**Review by candy02058912 — DISMISSED** [review]
Only blocking change is the Tracks event name

**Comment by crisbusquets** [comment]
About the responsiveness of the modal:

If it's **mobile**, we could use this approach (in this example, I'm using a 375px width):
* The change is in the layout (vertical instead of horizontal), margins (32px instead of 40px), and heading font size (28px instead of 32px).

* Figma file: swECLCmqZdAXDKUjGtPa8A-fi-397_9258
* Figma specs: swECLCmqZdAXDKUjGtPa8A-fi-397_9910

mobile

The modal for desktop should work even on small **tablets** (768px width):
tablet

**Comment by lsl** [comment]
Mobile design looks good, I'll do in a follow up, issue here https://github.com/Automattic/dotcom-forge/issues/5008

**Comment by lsl** [comment]
Bumped sizelimit since it was blocking the build, not sure if these sizes are hard limits or just a warning system for accidental blow outs. See https://github.com/Automattic/wp-calypso/pull/76914 for context, cc @jsnmoon

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10461415

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @lsl for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10461415).

**Pull request merged**
